tools/python/util/android/android.py

```diff
@@ -0,0 +1,151 @@
+        if is_windows():
+            return "{}.{}".format(name, windows_extension)
+        else:
+            return name
+
+    def resolve_path(dirnames, basename):
+        dirnames.insert(0, "")
+        for dirname in dirnames:
+            path = shutil.which(os.path.join(dirname, basename))
+            if path is not None:
+                path = os.path.realpath(path)
+                _log.debug("Found {} at {}".format(basename, path))
+                return path
+        _log.warning("Failed to resolve path for {}".format(basename))
+        return None
+
+    return SdkToolPaths(
+        emulator=resolve_path(
+            [os.path.join(sdk_root, "emulator")],
+            filename("emulator", "exe")),
+        adb=resolve_path(
+            [os.path.join(sdk_root, "platform-tools")],
+            filename("adb", "exe")),
+        sdkmanager=resolve_path(
+            [os.path.join(sdk_root, "tools", "bin"),
+             os.path.join(sdk_root, "cmdline-tools", "tools", "bin")],
+            filename("sdkmanager", "bat")),
+        avdmanager=resolve_path(
+            [os.path.join(sdk_root, "tools", "bin"),
+             os.path.join(sdk_root, "cmdline-tools", "tools", "bin")],
+            filename("avdmanager", "bat")))
+
+
+def create_virtual_device(
+        sdk_tool_paths: SdkToolPaths,
+        system_image_package_name: str,
+        avd_name: str):
+    run(sdk_tool_paths.sdkmanager, "--install", system_image_package_name,
+        input=b"y")
+
+    run(sdk_tool_paths.avdmanager, "create", "avd",
+        "--name", avd_name,
+        "--package", system_image_package_name,
+        "--force",
+        input=b"no")
+
+
+_process_creationflags = subprocess.CREATE_NEW_PROCESS_GROUP if is_windows() else 0
+
+
+def _start_process(*args) -> subprocess.Popen:
+    _log.debug("Starting process - args: {}".format([*args]))
+    return subprocess.Popen([*args], creationflags=_process_creationflags)
+
+
+_stop_signal = signal.CTRL_BREAK_EVENT if is_windows() else signal.SIGTERM
+
+
+def _stop_process(proc: subprocess.Popen):
+    _log.debug("Stopping process - args: {}".format(proc.args))
+    proc.send_signal(_stop_signal)
+
+    try:
+        proc.wait(30)
+    except subprocess.TimeoutExpired:
+        _log.warning("Timeout expired, forcibly stopping process...")
+        proc.kill()
+
+
+def _stop_process_with_pid(pid: int):
+    # not attempting anything fancier than just sending _stop_signal for now
+    _log.debug("Stopping process - pid: {}".format(pid))
+    os.kill(pid, _stop_signal)
+
+
+def start_emulator(
+        sdk_tool_paths: SdkToolPaths,
+        avd_name: str) -> subprocess.Popen:
+    with contextlib.ExitStack() as emulator_stack, \
+         contextlib.ExitStack() as waiter_stack:
+        emulator_process = emulator_stack.enter_context(
+            _start_process(
+                sdk_tool_paths.emulator, "-avd", avd_name,
+                "-partition-size", "2047",
+                "-memory", "4096",
```

Comment:
Seems the Android CI failed running out of storage, change this to 4096 instead or maybe make an optional param default to 2047 but can specify bigger size?

Reply:
-partition-size seems to have a maximum value of 2047 in my Windows environment. added a way to specify different values and increased the CI build's to 4096

---
In reply to: [556928829](https://github.com/microsoft/onnxruntime/pull/6317#discussion_r556928829) [](ancestors = 556928829)